pkg/subnet/subnet.go

```diff
@@ -15,6 +17,84 @@ type ipAndNetwork struct {
 // stubbable version
 var interfaceAddrs = net.InterfaceAddrs
 
+// IPv4 subnet key. Identifies a class B subnet
+type ipv4SubnetKey [2]byte
+
+// IPv6 subnet key. This is the 48-bit route and 16-bit subnet identifier. Identifies a 64 bit subnet.
+type ipv6SubnetKey [8]byte
+
+// AnalyzeIPs returns the ip networks needed to cover the given IPs with as
+// big mask as possible for each subnet. The analyze starts by finding all
+// subnets using a 16-bit mask for IPv4 and a 64 bit mask for IPv6 addresses.
+// Once the subnets are established, the mask for each one will be increased
+// to the maximum value that still masks all IPs that it was created for.
+func AnalyzeIPs(ips []net.IP) []*net.IPNet {
```

Comment:
1. I think `AnalyzeIPs` is a poor name, it doesn't communicate what it's analyzing for.  In the Telepresence 1 code the equivalent function is named `covering_cidrs`.
2. It's probably worth adding a comment about how this logic relates to the `covering_cidrs` logic from Tel 1.  If the answer is "This was written, but wasn't compared to the code in Tel 1", that's fine, but we should write it down.  FWIW, I'm pretty sure an external contributor has an open PR to improve this logic in Tel 1.

Reply:
The whole `subnet` package is completely removed later on when the need to figure out IP's and merge them into subnets is gone.